Fixed: EntityQuery queryCount is throwing error with distinct method

pwverma opened the pull request:
(OFBIZ-11294)

Added missing support for selectFields in queryCount and all the subsequent methods.

dixitdeepak commented on framework/entity/src/main/java/org/apache/ofbiz/entity/GenericDelegator.java:
```diff
@@ -1625,8 +1625,8 @@ public EntityListIterator findListIteratorByCondition(DynamicViewEntity dynamicV
     @Override
-    public long findCountByCondition(String entityName, EntityCondition whereEntityCondition,
-            EntityCondition havingEntityCondition, EntityFindOptions findOptions) throws GenericEntityException {
+    public long findCountByCondition(String entityName, EntityCondition whereEntityCondition, Set<String> fieldsToSelect
```
@verma-pawan I think we need to keep the existing method signature as well, and can pass null as fieldsToSelect to maintain backward compatibility.

pwverma replied:
Thanks, @dixitdeepak totally agree. It slips off from my mind, on it.

pwverma replied:
@dixitdeepak Done, please have look!

pwverma committed:
Fixed: EntityQuery queryCount is throwing error with distinct method
(OFBIZ-11294)

Added missing support for selectFields in queryCount and all the subsequent methods.
